include/i3_containers.hpp

```diff
@@ -154,6 +154,7 @@ namespace i3_containers
         std::optional<std::uint64_t> window;              /**< X11 window ID of the client inside the container.      */
         std::optional<x11_window> window_properties;      /**< X11 properties of the window.                          */
         bool is_urgent;                                   /**< Whether the container is (in)directly urgent or not.   */
+        std::vector<const char*> marks;                   /**< List of marks assigned to container.                   */
```

Comment:
We should use `std::string` instead of `char*` because __RAPIDJSON__ library is managing memory for strings that it parses. Once the parsing is done and __i3-ipc++__ returns `node` to the client, __RAPIDJSON__ will release its memory, leaving us with dangling `char*` pointer. Instead, we need to copy data from __RAPIDJSON__'s `char*` into `std::string` while `char*` is still valid during parsing.
```suggestion
        std::vector<std::string> marks;                   /**< List of marks assigned to container.                   */
```